com.unity.shadergraph/Editor/Drawing/Blackboard/SGBlackboardCategory.cs

```diff
@@ -0,0 +1,623 @@
+                {
+                    AddToClassList("unnamed");
+                }
+                else
+                {
+                    RemoveFromClassList("unnamed");
+                }
+            }
+        }
+
+        public bool headerVisible
+        {
+            get { return m_Header.parent != null; }
+            set
+            {
+                if (value == (m_Header.parent != null))
+                    return;
+
+                if (value)
+                {
+                    m_MainContainer.Add(m_Header);
+                }
+                else
+                {
+                    m_MainContainer.Remove(m_Header);
+                }
+            }
+        }
+
+        void SetDragIndicatorVisible(bool visible)
+        {
+            m_DragIndicator.visible = visible;
+        }
+
+        public bool CategoryContains(List<ISelectable> selection)
+        {
+            if (selection == null)
+                return false;
+
+            // Look for at least one selected element in this category to accept drop
+            foreach (ISelectable selected in selection)
+            {
+                VisualElement selectedElement = selected as VisualElement;
+
+                if (selected != null && Contains(selectedElement))
+                {
+                    if (canAcceptDrop == null || canAcceptDrop(selected))
+                        return true;
+                }
+            }
+
+            return false;
+        }
+
+        void OnFoldoutToggle(ChangeEvent<bool> evt)
+        {
+            if (evt.previousValue != evt.newValue)
+            {
+                var isExpandedAction = new ChangeCategoryIsExpandedAction();
+                if (selection.Contains(this)) // expand all selected if the foldout is part of a selection
+                    isExpandedAction.categoryGuids = selection.OfType<SGBlackboardCategory>().Select(s => s.viewModel.associatedCategoryGuid).ToList();
+                else
+                    isExpandedAction.categoryGuids = new List<string>() { viewModel.associatedCategoryGuid };
+
+                isExpandedAction.isExpanded = evt.newValue;
+                isExpandedAction.editorPrefsBaseKey = blackboard.controller.editorPrefsBaseKey;
+                viewModel.requestModelChangeAction(isExpandedAction);
+            }
+        }
+
+        internal void TryDoFoldout(bool expand)
+        {
+            m_Foldout.SetValueWithoutNotify(expand);
+            if (!expand)
+                m_RowsContainer.RemoveFromHierarchy();
+            else
+                m_MainContainer.Add(m_RowsContainer);
+
+            var key = $"{blackboard.controller.editorPrefsBaseKey}.{viewDataKey}.{ChangeCategoryIsExpandedAction.kEditorPrefKey}";
+            EditorPrefs.SetBool(key, expand);
+        }
+
+        void OnMouseDownEvent(MouseDownEvent e)
+        {
+            if ((e.clickCount == 2) && e.button == (int)MouseButton.LeftMouse && IsRenamable())
```

Comment:
Is this handling double click? A simple comment stating such might be nice

Reply:
will add 👍